server/src/main/java/org/elasticsearch/index/mapper/TsidExtractingIdFieldMapper.java

```diff
@@ -119,9 +122,9 @@ public static void createField(DocumentParserContext context, IndexRouting.Extra
          * it always must pass.
          */
         IndexRouting.ExtractFromSource indexRouting = (IndexRouting.ExtractFromSource) context.indexSettings().getIndexRouting();
-        assert context.getDynamicMappers().isEmpty() == false
-            || context.getDynamicRuntimeFields().isEmpty() == false
-            || id.equals(indexRouting.createId(TimeSeriesIdFieldMapper.decodeTsid(tsid), suffix));
+        // assert context.getDynamicMappers().isEmpty() == false
```

Comment:
I need to comment this assertion, at least for now, because when we hash the tsid we do not serialise dimensions **and routing fields** which later on results in a failure of the assertion since `createId` will throw an exception due to missing routing fields.

Reply:
Maybe this is a matter of where we do the hashing? Maybe we should alter `TimeSeriesIdBuilder` class.
Just hash the `timeSeriesId`  in `TimeSeriesIdFieldMapper#postParse(...)` after line 148 only if it is beyond the 32KB limit and store that as `SortedDocValuesField`. And just pass down the original id to `TsidExtractingIdFieldMapper#createField(...)`?

Reply:
Can this assertion be re-enabled?

Reply:
`indexRouting.createId` requires a Map as input... `decodeTsid` does not return a Map anymore.
I don't think we can have that assertion. Not `decodeTsid` returns a Base4 encoding of the tsid and I see no way to retrieve the id given a Base64 encoding of the tsid.